### PR TITLE
PE-636: Send community tip with the data tx or bundle tx

### PR DESCRIPTION
**Pull request by fedellen**
This PR adds the ArDrive community tip to the bundled transaction or to the data transaction in the case of `--no-bundle`.

Adding support for this was fairly straight forward. The only complication was how to handle the tip parameters on the bulk folder upload method. Given that my intent is to deprecate the bulk folder method and create a new `uploadAllEntities` method in the bulk bundle branch, I have temporarily made `uploadAsV2Tx` accessible and have used it directly to avoid the conflict.

The unnecessary tip tag `"Type"` has been removed during this PR in compliance with the [GQL Tag Usage Doc](https://ardrive.atlassian.net/wiki/spaces/ENGINEERIN/pages/194969618/Graph+Query+Language+Tags+and+Usage#Community-Tip-Transactions)

### CODE REVIEW

File: src/arfs/arfsdao.ts [hunk 1]
```diff
@@ -755,6 +787,15 @@ export class ArFSDAO extends ArFSDAOAnonymous {
 			txAttributes.last_tx = 'STUB';
 		}
 
+		if (communityTipSettings) {
+			// Add community tip to the bundle trx
```
Comment (fedellen) [inline src/arfs/arfsdao.ts:791]:
```suggestion
			// Add community tip to the v2 transaction
```

**Review by matibat — COMMENTED** [review]
It's looking good. Gotta review the CLI side

**Review by matibat — APPROVED** [review]
Looks good!

**Review by jdaev — APPROVED** [review]
Looks good

File: src/ardrive.ts [hunk 2]
```diff
@@ -483,10 +491,12 @@ export class ArDrive extends ArDriveAnonymous {
 					key: isPrivateResult(uploadFileResult) ? urlEncodeHashKey(uploadFileResult.fileKey) : undefined
 				}
 			],
-			tips: [tipData],
+			tips: [],
 			fees: {}
 		};
 
+		const tipResult = { recipient: communityTipTarget, winston: communityWinstonTip };
```
Comment (arielmelendez) [inline src/ardrive.ts:498]: Consider typing this as `Partial<TipData>` to indicate to the reader that it's an incomplete, typed object.

Comment (fedellen) [reply src/ardrive.ts]: Went with `Omit<TipData, 'txId'>` to keep the compiler happy, it didn't like the optionals: d09e1e0

File: src/arfs/arfsdao.ts [hunk 3]
```diff
@@ -635,16 +636,20 @@ export class ArFSDAO extends ArFSDAOAnonymous {
 		return { arFSObjects: [dataArFSObject, metaDataArFSObject], fileId };
 	}
 
-	private async uploadFileV2Tx(
+	// Temporarily public for bulk folder use case, which will be deprecated in PE-460
+	// Re-make this private once `uploadAllEntities` exists
+	/* private */ async uploadFileV2Tx(
```
Comment (arielmelendez) [inline src/arfs/arfsdao.ts:641]: Consider marking as deprecated immediately

**Review by arielmelendez — APPROVED** [review]

**Pull request merged**